src/util.cpp

```diff
@@ -732,14 +732,21 @@ void FileCommit(FILE *file)
 #ifdef WIN32
     HANDLE hFile = (HANDLE)_get_osfhandle(_fileno(file));
     FlushFileBuffers(hFile);
-#else
-    #if defined(__linux__) || defined(__NetBSD__)
-    fdatasync(fileno(file));
-    #elif defined(__APPLE__) && defined(F_FULLFSYNC)
+#elif defined(__APPLE__) && defined(F_FULLFSYNC)
     fcntl(fileno(file), F_FULLFSYNC, 0);
-    #else
+#elif HAVE_FDATASYNC
+    fdatasync(fileno(file));
+#else
+    fsync(fileno(file));
```

Comment:
You aren't changing this, but it would be nice if these functions returned errors so failures could be logged.